### PR TITLE
Improved DSC service and added unit tests

### PR DESCRIPTION
**Pull request by DirvaNIcolae**

### CODE REVIEW

**Review by alexandrudoncea — APPROVED** [review]

File: src/uds/diagnostic_session_control/include/DiagnosticSessionControl.h [hunk 1]
```diff
@@ -1,3 +1,18 @@
+/**
+ * @file DiagnosticSessionControl.h
+ * @author your name (you@domain.com)
```
Comment (RubenNatanael) [inline src/uds/diagnostic_session_control/include/DiagnosticSessionControl.h:3]: Add your name here

File: src/uds/diagnostic_session_control/src/DiagnosticSessionControl.cpp [hunk 2]
```diff
@@ -24,16 +24,16 @@ DiagnosticSessionControl::~DiagnosticSessionControl()
 }
 
 /* Method to control the sessions of service */
-void DiagnosticSessionControl::sessionControl(uint8_t id, uint8_t sub_function)
+void DiagnosticSessionControl::sessionControl(canid_t frame_id, uint8_t id, uint8_t sub_function)
 {
     uint8_t request[] = {id, sub_function};
```
Comment (RubenNatanael) [inline src/uds/diagnostic_session_control/src/DiagnosticSessionControl.cpp:29]: You added 'id' variable in to 'request' array, but the 'id' variable is never used. Please try to minimize the number of variables, and use just the necessary ones. Delete id variable from the function's parameter

Comment (DirvaNIcolae) [reply src/uds/diagnostic_session_control/src/DiagnosticSessionControl.cpp:29]: frame_id is needed in the sessionControl function to send it as a parameter in the handleRequest function, where it is also given as a parameter in the _switch_ functions.

File: src/uds/diagnostic_session_control/include/DiagnosticSessionControl.h (hunk 1 above)
Comment (DirvaNIcolae) [reply src/uds/diagnostic_session_control/include/DiagnosticSessionControl.h]: done

File: src/uds/diagnostic_session_control/src/DiagnosticSessionControl.cpp (hunk 2 above)
Comment (RubenNatanael) [reply src/uds/diagnostic_session_control/src/DiagnosticSessionControl.cpp:29]: I'm not talking about 'frame_id' but the variable 'id' that you put in an array called request, that you pass on to handlerRequest, and rename it as sid, but it's not used anywhere

Comment (DirvaNIcolae) [reply src/uds/diagnostic_session_control/src/DiagnosticSessionControl.cpp]: Correct. I removed the `id`.

**Review by RubenNatanael — APPROVED** [review]

**Comment by alexandrudoncea** [comment]
please resolve the conflicts

**Pull request merged**